modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionsExchangeFuture.java

```diff
@@ -2350,6 +2379,23 @@ private String exchangeTimingsLogMessage(String header, List<String> timings) {
         return false;
     }
 
+    /**
+     * Updates the {@link GridMetricManager#CACHE_OPERATIONS_BLOCKED_DURATION_HISTOGRAM} metric if needed.
+     *
+     * @param duration The total duration of the current PME.
+     */
+    private void updateDurationHistogram(long duration) {
+        if (isClientEventExchangeWithoutAffinityChange())
+            return;
+
+        MetricRegistry mreg = cctx.kernalContext().metric().registry(GridMetricManager.SYS_METRICS);
```

Comment:
Can we store this metric in the class field instead of searching it on each PME?